lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -758,6 +758,40 @@ OpFoldResult AndOp::fold(ArrayRef<Attribute> constants) {
   return constFoldAssociativeOp(constants, hw::PEO::And);
 }
 
+/// returns a single common operand that all inputs of the operation `op` can
+/// be traced back to, or an empty `Value` if no such operand exists.
+///
+/// for example for `or(a[0], a[1], ..., a[n-1])` this function returns `a`
+/// (assuming the bit-width of `a` is `n`)
+template <typename Op>
+static Value getCommonOperand(Op op) {
+  auto inputs = op.inputs();
+
+  auto commonOperand = [&inputs](size_t index) -> Value {
+    if (auto extractOp = inputs[index].template getDefiningOp<ExtractOp>()) {
+      return extractOp.getOperand();
+    } else {
+      return Value();
+    }
+  };
+
+  Value source = commonOperand(0);
+
+  size_t size = inputs.size();
+
+  if (!source || size != source.getType().getIntOrFloatBitWidth()) {
+    return Value();
+  }
```

Comment:
Please remove braces on single statement if here and other places. 

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements